lib/reek/ast/node.rb

```diff
@@ -102,6 +104,9 @@ def source
         loc.expression.source_buffer.name
       end
 
+      # Method will be overridden by the code in the IfNode, CaseNode, and LogicOperatorBase sexp extensions.
+      def condition; end
```

Comment:
As I've been working on RuboCop node extensions, I've sometimes come across cases where having a `NullNode` abstraction would simplify the implementation of cops. I'm still not sure if this will be more help or hinderance though. Have you come across cases where this would be helpful? 🙂 

This is a pretty non-specific question, so I'm okay with an equally non-specific answer. 😬

Reply:
Thanks for the input @mvz! One of the downsides I've considered is it could cause code that should (rightfully) fail horribly to instead sort of pass. In RuboCop, we have *a lot* of:

```ruby
if node && node.foo_type?
```

so my first, primitive implementation would be just a node responding `false` to `#*_type?`.

Reply:
Yeah, you would have to be very careful with how it interacts with existing code.